.github/workflows/build_linux.yml

```diff
@@ -137,51 +141,228 @@ jobs:
         run: |
           cmake --build build -t assign_cpp_examples
 
-      - name: Compile tests as cpp code
-        run: |
-          cmake --build build -t all_tests_compile_as_cpp_code
-
-      - name: Compile tests as circuits
+      - name: Copy examples' circuits and assignments for uploading
         run: |
-          cmake --build build -t all_tests_compile_as_circuits
+          chmod +x ./examples/copy_artifacts.sh
+          ./examples/copy_artifacts.sh build/examples/cpp examples_output
 
-      - name: Run tests as cpp code (expected res calculation)
+      - name: Set aritfacts' names
+        id: artifact-names
         run: |
-          cmake --build build -t all_tests_run_expected_res_calculation -j$(nproc)
+          echo "transpiler=transpiler-output" >> $GITHUB_OUTPUT
+          echo "examples=circuits-and-assignments" >> $GITHUB_OUTPUT
 
-      - name: Run tests as circuits (real res calculation)
-        run: |
-          cmake --build build -t all_tests_assign_circuits -j$(nproc)
-
-      - name: Compare expected and real test results
+      - name: Upload examples' circuits and assignments artifact
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ steps.artifact-names.outputs.examples }}
+          path: |
+            examples_output
+
+      # - name: Run tests pallas
+      #   run: |
+      #     chmod +x ./tests/run_tests.sh
+      #     bash ./tests/run_tests.sh pallas -j$(nproc)
+
+      # - name: Run tests bls12381
+      #   run: |
+      #     chmod +x ./tests/run_tests.sh
+      #     bash ./tests/run_tests.sh bls12381 -j$(nproc)
+
+      # - name: Run tests on faulty inputs
+      #   run: |
+      #     chmod +x ./tests/run_tests_faulty_input.sh
+      #     bash ./tests/run_tests_faulty_input.sh pallas -j$(nproc)
+      - name: Get transpiler output for integration testing
         run: |
-          chmod +x ./tests/run_test_results_comparison.sh
-          bash ./tests/run_test_results_comparison.sh
+          cmake --build build \
+            -t arithmetics_cpp_example_evm_verifier \
```

Comment:
Can we put all the test names in one place in the script or get some other way (with a prefix maybe)? We duplicate them several times in the script.